### PR TITLE
Created Review.java

### LINKED ISSUE #7: [SUBTASK] Creating a Reviews Table
**What feature is this a subtask of**
This subtask is part of Feature #3.

**What will be implemented in this subtask**
This subtask implements the backend.
  
**What is the acceptance criteria of this subtask**
This task is done when the review DAO is able to access the table easily.

**What files will be created/modified**
List the html/css/js/java/etc. files that you think will need to be modified or created (and where they are).

**What dependencies does this issue have or create**
The DAO cannot be created until after the table is made (see #4) 

**What is the estimated effort of this subtask**
The estimated effort of this subtask should be 2 to 3 days at most.

### PR DESCRIPTION
**Pull request by jding712**
## Issue
Issue resolves #7

## Description
The purpose of the PR is to create a java file for the reviews table.

## Files changed
The created file is Review.java and this file includes all the columns and parameters.

## Notes for reviewers
These are all the following parameters that were created for the table:
- user ID
- document ID
- reviewer ID
- rating
- creation date

## Checklist
Fill out this checklist before merging the PR:
- [X] Created issue
   - [X] Issue has a relevant labels
   - [x] Issue added to the kanban board with its status as In-Progress
   - [x] Issue added to the relevant milestone
   - [x] Issue mentioned in the first section of this PR description
- [x] Testing
   - [x] Added new tests for the changes (or modified existing tests)
   - [x] Passed most recent round of CI
- [x] Review
   - [x] Received first round of review comments
   - [x] Addressed first round of review comments
   - [x] Received second round of review comments
   - [x] Addressed second round of review comments

### CODE REVIEW

File: docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java
Comment (knassre-cmu) [inline docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java]: An open question: do we want to show the user's names or IDs? I don't have a good answer. In the meantime, just to hedge our bets, can we add the names also to the table?

In a real database application, we'd just call INNER JOIN on the User table to obtain their names, but we are assuming for the purposes of this feature that all of the users in the table only exist in the this table (i.e. we have not added them to the User table).

File: docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java
Comment (knassre-cmu) [inline docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java]: For consistency, I would call it `rating` instead of `rat`

File: docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java
Comment (knassre-cmu) [inline docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java]: For consistency, I would call the argument `reviewerId` instead of `id`

File: docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java
Comment (knassre-cmu) [inline docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java]: For consistency, I would call the argument `docId` instead of `id`

**Review by knassre-cmu — DISMISSED** [review]
@jding712 Well done! It looks good to me. I have 3 minor comments and one important one, but you should be able to resolve those ahead of 2nd so you won't need to push back to me for approval.

File: docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java
Comment (knassre-cmu) [inline docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java]: One more thing: let's add another column that represents the ID of the review itself (just like each user, document, acl, etc. has its own ID). That one should be the one with the @Id decorator (i.e. a primary key).

Comment (knassre-cmu) [inline docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java]: Shoot almost forgot this: Need to add `implements Loggable` for compatibility reasons

File: docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java
Comment (aruiz2) [inline docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java]: I don't really understand what the function does based on this description. Do you think you could explain it in depth such that a person that is not very familiar with the codebase could understand it better?

File: docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java
Comment (aruiz2) [inline docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java]: "REV_REVIEWERID_C" might be better since then variable in the next line is called "reviewerID"

File: docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java
Comment (aruiz2) [inline docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java]: Should this not be "setReviewerRating" ?

**Review by aruiz2 — APPROVED** [review]

File: docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java
Comment (knassre-cmu) [inline docs-core/src/main/java/com/sismics/docs/core/model/jpa/Review.java:10]: TODO:
- Add file `dbupdate-028-0.sql` in the folder `docs-core/src/main/resources/db/update` with the SQL code to create the SQL table
- Update all three `config.properties` files to say `db.version=28`

**Review by knassre-cmu — COMMENTED** [review]
Left comments for the major revisions required based on what we heard from profs

File: docs-core/src/main/resources/db/update/dbupdate-028-0.sql
Comment (aruiz2) [inline docs-core/src/main/resources/db/update/dbupdate-028-0.sql:2]: Should we make the "REV_USERNAME_C", "REV_DOCID_C", "REV_REVIEWERID_C" to be foreign keys?

File: docs-core/src/main/resources/db/update/dbupdate-028-0.sql
Comment (knassre-cmu) [inline docs-core/src/main/resources/db/update/dbupdate-028-0.sql]: Delete date should not be `not null`, since it is null until the entry is deleted

File: docs-core/src/main/resources/db/update/dbupdate-028-0.sql
Comment (knassre-cmu) [reply docs-core/src/main/resources/db/update/dbupdate-028-0.sql:2]: Ehhh... I vote let's avoid the headache for the sake of our collective convenience.

File: docs-web/src/prod/resources/config.properties
Comment (knassre-cmu) [inline docs-web/src/prod/resources/config.properties:3]: I think there is one more of these (one in docs-core)

**Review by knassre-cmu — COMMENTED** [review]
Left some comments. After resolution I'll approve :)

**Review by knassre-cmu — APPROVED** [review]
LGTM

**Review by aruiz2 — APPROVED** [review]

**Pull request merged**
